src/core/mempool.rs

```diff
@@ -1856,13 +1905,35 @@ impl MemPoolDB {
                         test_debug!("No more txs after {}", &next_tx.txid());
                         break;
                     }
-                } else {
-                    // no more
+                } else if let Some(next_txid) = next_last_randomized_txid_opt {
+                    test_debug!("No rows returned for {}", &query.last_randomized_txid);
+
+                    // no rows found
+                    query.last_randomized_txid = next_txid;
+
+                    // send the next page ID
+                    query.tx_buf_ptr = 0;
+                    query.tx_buf.clear();
+                    query.corked = true;
+
                     test_debug!(
-                        "No more txs in query after {:?}",
+                        "Cork tx stream with next page {}",
                         &query.last_randomized_txid
                     );
-                    break;
+                    query
+                        .last_randomized_txid
+                        .consensus_serialize(&mut query.tx_buf)
+                        .map_err(ChainstateError::CodecError)?;
```

Comment:
According to code coverage, this branch isn't covered -- is it possible to add coverage for this to the unit tests?